### ISSUE TITLE
Variational Autoencoder does not converge with tensorflow on CPU

### ISSUE THREAD

**Issue by dhaval067** [keras-team/keras#7800]
I am facing strange issue on running variational autoencoder example. 

Keras version used: 2.0.8

With Tensorflow backend:
Tensorflow version: 1.3.0 (also checked on 1.2.1)
On GPU code gives the perfect results (i.e. loss getting converse) but when i try to run it on CPU loss value of the function getting diverse. After around 11 ephochs loss value starts to increase. 

With Theano backend:
It is working perfectly with CPU.

Why this type of uncertain behaviour happening from same code?

Any one please help me with this.

@fchollet can you look into this?

**Comment by dbl001** [comment]
Same problem with Tensorflow version 1.2.0.

**Comment by lxkain** [comment]
I can confirm this also.

**Comment by lxkain** [comment]
actually, I got it to work by setting epsilon_std = 0.01
